internal/core/context.go

```diff
@@ -36,9 +36,9 @@ const (
 	metaContextKey contextKey = iota
 )
 
-// newMetaContext creates a new ctx with injected meta and returns it.
-func newMetaContext(meta *meta) context.Context {
-	return context.WithValue(context.Background(), metaContextKey, meta)
+// injectMeta creates a new ctx with injected meta and returns it.
```

Comment:
nit: modifies the given context

Reply:
Context are imutable object in Go. `WithValue` create a new child context of the on given as parameter

Reply:
Yep but it's based on the context passed to the function

Reply:
Something like `creates a new ctx based on the given one in the arguments`